src/EditorFeatures/CSharpTest/ExtractMethod/ExtractMethodTests.cs

```diff
@@ -11336,5 +11336,125 @@ private static IntPtr NewMethod(object value)
 ";
             await TestExtractMethodAsync(code, expected);
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ExtractMethod)]
+        public async Task ExtractRawStringLiteral_SingleLine()
+        {
+            var code = @"
+class C
+{
+    void M(int y)
+    {
+        var s = [|""""""Hello world""""""|];
+    }
+}";
+            var expected = @"
+class C
+{
+    void M(int y)
+    {
+        var s = GetS();
+    }
+
+    private static string GetS()
+    {
+        return """"""Hello world"""""";
+    }
+}";
+
+            await TestExtractMethodAsync(code, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ExtractMethod)]
+        public async Task ExtractRawStringLiteralInterpolation_SingleLine()
+        {
+            var code = @"
+class C
+{
+    void M(int y)
+    {
+        var s = [|$""""""{y}""""""|];
+    }
+}";
+            var expected = @"
+class C
+{
+    void M(int y)
+    {
+        var s = GetS(y);
+    }
+
+    private static string GetS(int y)
+    {
+        return $""""""{y}"""""";
+    }
+}";
+
+            await TestExtractMethodAsync(code, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ExtractMethod)]
+        public async Task ExtractRawStringLiteral_MultiLine()
+        {
+            var code = @"
+class C
+{
+    void M(int y)
+    {
+        var s = [|""""""
+            Hello world
+            """"""|];
+    }
+}";
+            var expected = @"
+class C
+{
+    void M(int y)
+    {
+        var s = GetS();
+    }
+
+    private static string GetS()
+    {
+        return """"""
+            Hello world
+            """""";
+    }
+}";
+
+            await TestExtractMethodAsync(code, expected);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ExtractMethod)]
+        public async Task ExtractRawStringLiteralInterpolation_MultiLine()
+        {
+            var code = @"
+class C
+{
+    void M(int y)
+    {
+        var s = $[|""""""
+            {y}
+            """"""|];
+    }
+}";
+            var expected = @"
+class C
+{
+    void M(int y)
+    {
+        var s = GetS(y);
+    }
+
+    private static string GetS(int y)
+    {
+        return $""""""
+            {y}
+            """""";
+    }
+}";
+
+            await TestExtractMethodAsync(code, expected);
```

Comment:
one more tests with `$$"""` where there is a `{` as content as well as a brace plz.  (if you added this already, then that's great).

Note: the test should have all the braces next to each other as well.